### PR TITLE
bump zgw-consumers to 0.35.1, commonground-api-common to 2.0.1, OAF to 0.9.0

### PR DESCRIPTION
**Pull request by annashamray**
fixes https://github.com/maykinmedia/open-api-framework/issues/66

depends on https://github.com/maykinmedia/commonground-api-common/pull/45

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ RUN apt-get update && apt-get upgrade -y && apt-get install -y --no-install-reco
         pkg-config \
         build-essential \
         # only relevant when using editable/github dependencies, which is discouraged
-        # git \
+        git \
```
Comment (SonnyBA) [inline Dockerfile:14]: This should be reverted whenever a new release happens for `commonground-api-common`/ OAF.

File: requirements/base.in [hunk 2]
```diff
@@ -1,3 +1,4 @@
 open-api-framework
+git+https://github.com/maykinmedia/commonground-api-common.git@issue/44-migration-error
```
Comment (SonnyBA) [inline requirements/base.in:2]: Like the comment above, adding a reminder here that this should be pinned to a release version. We probably also want to get rid of pinning `commonground-api-common` in all the projects and pulling that dependency in through OAF, [see discussion](https://github.com/maykinmedia/open-api-framework/issues/66#issuecomment-2500098180).

**Review by SonnyBA — APPROVED** [review]
Looks good! This depends on a new release of `commonground-api-common` and probably OAF though (so that dependencies can be pinned properly).

**Review by bart-maykin — APPROVED** [review]

File: requirements/base.in (hunk 2 above)
Comment (SonnyBA) [reply requirements/base.in]: See https://github.com/maykinmedia/open-api-framework/issues/90

**Pull request merged**